packages/js/product-editor/src/blocks/product-fields/shipping-dimensions/edit.tsx

```diff
@@ -16,24 +15,21 @@ import {
 	useState,
 } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
-import classNames from 'classnames';
-import {
-	BaseControl,
-	// @ts-expect-error `__experimentalInputControl` does exist.
-	__experimentalInputControl as InputControl,
-} from '@wordpress/components';
 
 /**
  * Internal dependencies
  */
 import { ShippingDimensionsBlockAttributes } from './types';
-import { useProductHelper } from '../../../hooks/use-product-helper';
 import {
 	HighlightSides,
 	ShippingDimensionsImage,
 } from '../../../components/shipping-dimensions-image';
 import { useValidation } from '../../../contexts/validation-context';
 import { ProductEditorBlockEditProps } from '../../../types';
+import { NumberControl } from '../../../components/number-control';
+
+const SHIPPING_AND_WEIGHT_MIN_VALUE = 0;
+const SHIPPING_AND_WEIGHT_MAX_VALUE = 100000;
```

Comment:
This should deff be bigger, because these units can also be in `mm` or `grams`

Maybe go with `100_000_000_000_000`? Javascript supports formatted numbers right?

Reply:
> Javascript supports formatted numbers right?

didn't knew about this, but apparently it does! TIL.